src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/IVsHierarchyItemExtensions.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.RegularExpressions;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections
+{
+    /// <summary>
+    /// Extension methods for <see cref="IVsHierarchyItem"/> that support attached collections.
+    /// </summary>
+    internal static class IVsHierarchyItemExtensions
+    {
+        private static Regex? s_targetFlagsRegex;
+        private static Regex? s_packageFlagsRegex;
+
+        /// <summary>
+        /// Detects the configuration string associated with a given hierarchy item in the dependencies tree, if
+        /// nested within a target group node. For projects that do not multi-target, this will always return <see langword="false"/>.
+        /// This method searches ancestors until a configuration is found, or the project root is found.
+        /// </summary>
+        /// <param name="item">The item to test for.</param>
+        /// <param name="configuration">The detected configuration, if found.</param>
+        /// <returns><see langword="true"/> if the configuration was found, otherwise <see langword="false"/>.</returns>
+        public static bool TryFindConfiguration(
```

Comment:
This looks like it finds a dimension, not a configuration.

Reply:
It's the target configuration. Would you call this `TryFindTarget` instead?

Reply:
@davidwengier about a week ago I mentioned how the `$TFM:Foo` flag was not consumed by the Project System. FYI this PR changes that, and introduces new special flags for package ID/version.

Reply:
What makes this a "configuration"? It's returns the value of the "target framework" dimension, which is a part of the configuration.